packages/js/product-editor/src/components/attribute-term-input-field/attribute-term-input-field.tsx

```diff
@@ -278,24 +315,6 @@ export const AttributeTermInputField: React.FC<
 					);
 				} }
 			</SelectControl>
-			{ addNewAttributeTermName && attributeId !== undefined && (
-				<CreateAttributeTermModal
-					initialAttributeTermName={ addNewAttributeTermName }
-					onCancel={ () => {
-						setAddNewAttributeTermName( undefined );
-						focusSelectControl();
-					} }
-					attributeId={ attributeId }
-					onCreated={ ( newAttribute ) => {
-						onSelect( newAttribute );
-						setAddNewAttributeTermName( undefined );
-						invalidateResolutionForStoreSelector(
-							'getProductAttributeTerms'
-						);
-						focusSelectControl();
-					} }
-				/>
-			) }
```

Comment:
I don't think we should remove this Term Modal just yet, and instead leave this configurable.
Maybe add a prop -> `autoCreateOnSelect` and when that is `true` we don't show the Modal and auto create it.

Reply:
I added the modal back in the commit [933ece8](https://github.com/woocommerce/woocommerce/pull/39610/commits/933ece88b85dad0400b2ba44e5e295a11d48191d)